cypress/integration/parallel-2/braze.spec.js

```diff
@@ -0,0 +1,135 @@
+
+const idapiResponse = `{ "status": "ok", "user": { "primaryEmailAddress": "user@example.com", "id": "000000000", "publicFields": { "displayName": "user" }, "privateFields": { "brazeUuid": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa", "puzzleUuid": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "googleTagId": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "legacyPackages": "CRE,RCO", "legacyProducts": "CRE,RCO" }, "statusFields": { "userEmailValidated": true }, "dates": { "accountCreatedDate": "2021-01-01T00:00:00Z" }, "userGroups": [ { "path": "/sys/policies/basic-identity", "packageCode": "CRE" }, { "path": "/sys/policies/basic-community", "packageCode": "RCO" } ], "socialLinks": [ { "socialId": "111111111111111111111", "network": "google" } ], "adData": {}, "consents": [ { "actor": "user", "id": "sms", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "post_optout", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "phone_optout", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "market_research_optout", "version": 0, "consented": true, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "supporter", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "jobs", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "holidays", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "events", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "offers", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 }, { "actor": "user", "id": "profiling_optout", "version": 0, "consented": false, "timestamp": "2021-01-01T00:00:00Z", "privacyPolicyVersion": 1 } ], "hasPassword": false } }`;
+
+const handleGuCookieError = () => {
+	cy.on('uncaught:exception', (err, runnable) => {
+		// When we set the `GU_U` cookie this is causing the commercial bundle to try and do
+		// something with the url which is failing in Cypress with a malformed URI error
+		expect(err.message).to.include('URI malformed');
+		// This error is unrelated to the test in question so return  false to prevent
+		// this commercial error from failing this test
+		return false;
+	});
+};
+
+describe('Braze messaging', function () {
+	beforeEach(function () {
+		cy.clearLocalStorage();
+		handleGuCookieError();
+		setLocalBaseUrl();
+	});
+
+	const visitArticle = (
+		url = 'https://m.code.dev-theguardian.com/games/2018/aug/23/nier-automata-yoko-taro-interview',
```

Comment:
Hey good spot. Yep it is intentional, to ensure that we get configuration for the code env. For example the Braze API key for that env, rather than the prod one. It feels more correct to be interacting with the code Braze account than the prod one.